library/src/dotty/runtime/LazyVals.scala

```diff
@@ -20,24 +18,45 @@ object LazyVals {
         }
       }
 
+  private[this] val base: Int = {
+    val processors = java.lang.Runtime.getRuntime.availableProcessors()
+    8 * processors * processors
+  }
+  private[this] val monitors: Array[Object] =
+    Array.tabulate(base)(_ => new Object)
```

Comment:
```scala
Array.fill(base)(new Object)
```
?

Reply:
I really don't like `CollectionName.fill`. It doesn't make the intent very clear. When I read `Array.fill(base)(new Object)`, to me it looks like the array is filled with the same instance of `Object`.  I really think it is an abuse of by-name parameters. I feel like by-name parameters should really have a "call by need" semantic 😄 

That said, it is my personal opinion and I am happy to change it if you feel strongly about it.

Reply:
Fair enough. I understand that people don't like it ;) Let's keep `tabulate`, then.